mitodl/ocw-studio
# allow overriding hugo build arguments

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1664

#### What's this PR do?
This PR adds an argument to the management commands `backpopulate_pipelines` as well as `upsert_mass_build_pipeline` called `hugo_args` that allows you to pass a string of arguments to append / override in the Hugo commands used by pipelines. While this PR gives you a method of bypassing the issues faced in https://github.com/mitodl/ocw-studio/issues/1664, it is more generally useful in that while developing locally and testing different things with Hugo builds in Concourse pipelines you can arbitrarily change the arguments passed to `hugo`.

Since we have decided to not pursue unifying the online and offline builds, we reverted the `course-v3` and `course-offline-v2` themes in https://github.com/mitodl/ocw-hugo-themes/pull/1081 and https://github.com/mitodl/ocw-hugo-projects/pull/245, therefore the issue in https://github.com/mitodl/ocw-studio/issues/1664 cannot be reproduced on the `main` branches of `ocw-hugo-themes` and `ocw-hugo-projects`.

#### How should this be manually tested?
Since https://github.com/mitodl/ocw-hugo-themes/pull/1081 and https://github.com/mitodl/ocw-hugo-projects/pull/245 have been reverted, we cannot exactly reproduce the issue from https://github.com/mitodl/ocw-studio/issues/1664 so we will verify that everything is working by enabling the `--templateMetrics` flag in a single course build:

 - Spin up `ocw-studio` with `docker compose up`, making sure you have set up Concourse and Minio as described in the readme
 - Run `docker compose exec web ./manage.py backpopulate_pipelines --filter 10-302-transport-processes-fall-2004 --hugo_args="--templateMetrics"`, replacing the filter string with the course of your choice if desired
 - Publish the course from the `ocw-studio` UI
 - Go to the Concourse web UI at http://concourse:8080 and look for the pipeline matching the site you published by putting the ID in the search box
 - Open the `build-course-offline` or `build-course-online` steps and verify that you see template metrics in the terminal output
